### DISCUSSION TITLE
How to switch back to the production Firebase after connecting a RN app to the Firebase emulator

### DISCUSSION THREAD

**Discussion by hayata-suenaga** [invertase/react-native-firebase#7848] [Q&A] [answered]
You can connect your development app to the Firebase emulator running on your development device by using `firestore().useEmulator(origin, 8080)` or `auth().useEmulator(`http://${origin}:9099`)`.

However, reverting to the production instance is not straightforward. The app appears to keep communicating with the Emulator instead of the production Firebase, even when the `useEmulator` invocations are commented out. A potential workaround is to uninstall the development client app and reinstall it.

The issue might arise because the default Firebase app instance, once configured to connect to the emulator, continues to do so even after the app restarts.

I attempted to delete the default app instance and reinitialize it, but deleting the default Firebase app instance seems to be impossible.

Does anyone have a solution for smoothly switching between the production and emulator instances?

**Comment by Ren-Logronio** [comment]
<img width="318" height="385" alt="image" src="https://github.com/user-attachments/assets/5a5d2d70-a674-4433-a22d-fa3abd859611" />

You can erase the data from the ios simulator but you would have to make the development build again. Idk what is the equivalent of this for the android emulator (maybe clearing the data is enough).

But the fact that there is still no tool and solution for this is depressing, its been 1 year and the problem is still there.

**Comment by mikehardy** [comment, marked as answer]
I don't see any problem with it and in fact use cloud auth and emulator auth all the time.
I just used it extensively over the last week in order to manually verify the TOTP PR we just merged since the emulator doesn't support it and we had to use cloud, but we use the emulator for the rest of our e2e testing.

In my experience it is absolutely as straightforward as: if you want to use the emulator, make those useEmulator / connectEmulator etc API calls, if you do not want to use it, do not make those calls.

You will need to post a reproduction that demonstrates it does not work, but I'll beat you to that and post a reproduction that does work: our e2e test app which now has some manual testing panels

`yarn && yarn tests:android:build` to build it
`yarn tests:packager:jet` to start the packager, leave this running
`yarn tests:emulator:start` to start the firebase emulator, leave this running
`yarn tests:android:test` to run the e2e tests. they use the emulator.
`yarn tests:android:manual` to run e2e test app in manual mode, open the manual tests and use the TOTP test. It uses cloud auth.

I see no problem. Can do same for macos/other, and ios. The all work.

Will reopen if you can post a minimum complete reproducer

**Comment by harelw** [comment]
I am seeing this behavior with the current version of Expo (expo@54.0.29). @mikehardy what's the ideal way to supply a repro?

`npm run ios` is the fastest way to resolve it (stop connecting to emulator, start connecting to configured firebase env)
